### ISSUE TITLE
Results: Interpretation and Visualization

### ISSUE THREAD

**Issue by FleetingA** [ultralytics/yolov5#4269]
I'm using YOLOv5 on my custom data. I have a single class of objects. I finished training and ran val.py to check model accuracy at this stage.

I have two questions:

1) How to interpret the resulting confusion matrix (i.e., absence of values)?

![confusion_matrix](https://user-images.githubusercontent.com/78978040/127835480-10bd0939-6f49-4fa9-9ce0-50f7ab9e2c16.png)

2) I ran val.py on a full folder of images, but the runs > val subfolder only contains batches 0, 1 and 2. Is there a way to see how the algorithm has performed on all the validation images (without running val.py on every single one separately)?

Thanks in advance!

**Comment by glenn-jocher** [comment]
@FleetingA appears to be a problem with your validation data. val.py naturally runs on all validation data. Follow Train Custom Data tutorial below for instructions on dataset setup.

## YOLOv5 Tutorials

* [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data)&nbsp; 🚀 RECOMMENDED
* [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)&nbsp; ☘️ RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Supervisely Ecosystem](https://docs.ultralytics.com/yolov5/tutorials/model_export8)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training)
* [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading)&nbsp; ⭐ NEW
* [TorchScript, ONNX, CoreML Export](https://docs.ultralytics.com/yolov5/tutorials/model_export) 🚀
* [Test-Time Augmentation (TTA)](https://docs.ultralytics.com/yolov5/tutorials/test_time_augmentation)
* [Model Ensembling](https://docs.ultralytics.com/yolov5/tutorials/model_ensembling)
* [Model Pruning/Sparsity](https://docs.ultralytics.com/yolov5/tutorials/model_pruning_and_sparsity)
* [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution)
* [Transfer Learning with Frozen Layers](https://docs.ultralytics.com/yolov5/tutorials/transfer_learning_with_frozen_layers)&nbsp; ⭐ NEW
* [TensorRT Deployment](https://github.com/wang-xinyu/tensorrtx)

**Comment by FleetingA** [comment]
Thank you, when it comes to val.py naturally running on all validation data, I meant: is there a way to see which images were true positives, false positives, false negatives, etc.? Or should one go through images one by one in order to obtain this information?

**Comment by glenn-jocher** [comment]
@FleetingA on an individual image level you can observe val_batch labels and predictions jpgs to compare. These are always plotted to your save directory, i.e. runs/val/exp

**Comment by FleetingA** [comment]
Thanks, perhaps I am not explaining myself very well. My point was whether
there is a way to know which of the predictions are correct, which are
false positives, false negatives, etc. What we get are predicted bounding
boxes and their associated probabilities, but is there a way to pick out
FPs, FNs, etc.? Not sure if my question makes sense.

**Comment by glenn-jocher** [comment]
@FleetingA you can view **labels** and predictions side by side as I mentioned in https://github.com/ultralytics/yolov5/issues/4269#issuecomment-891170769
